### ISSUE TITLE
SDL2-2.0.22 crashes client

### ISSUE THREAD

**Issue by biopsin** [etlegacy/etlegacy#1978]
I updated from SDL2-2.0.20 to SDL2-2.0.22 and the client crashes upon launching with just message
```
0 ----- Client Shutdown ----------
0 RE_Shutdown( 1 )
0 SDL input devices shut down.
0 --------------------------------
```
With version 2.0.20 output is
```
0 ----- Initializing Sound (1) ---
0 SDL_Init( SDL_INIT_AUDIO )... OK
0 SDL audio driver is "alsa".
0 SDL_AudioSpec:
0   Format:   AUDIO_S16LSB
0   Freq:     44100
0   Samples:  1024
0   Channels: 2
0   Silence:  0
0   Size:     4096
0 Starting SDL audio callback...
0 SDL audio initialized.
0 ----- Sound Info -----
0 sound system is muted
0     2 channels
0 32768 samples
0    16 samplebits
0     1 submission_chunk
0 44100 speed
0 0x55941e3f9080 dma buffer
0 No background file.
0 ----------------------
0 Sound initialization successfully done
s_backend set to SDL2
0 --------------------------------
```
Oh yea I did rebuild against new version just to make shure.
Linux voidx 5.4.190_1 #1 SMP PREEMPT 1650694802 x86_64 GNU/Linux

**Comment by Aciz** [comment]
We're still using SDL 2.0.20 with our releases. I just upgraded my system to SDL2.0.22, but since the client is still using 2.0.20, everything is working fine. Are you compiling yourself without bundled SDL2?

**Comment by biopsin** [comment]
Yes, rolling distro but I dont mind holding back as etlegacy is the only one depending on SDL2 on my system.

**Comment by Aciz** [comment]
I see, well I'll keep this open as a reminder for when we're eventually gonna upgrade to 2.0.22.

**Comment by Aciz** [comment]
The crash happens here on `glTexImage2D` call when the renderer is trying to create built-in images (specifically `tr.whiteImage`, but I don't think it's relevant, it's simply the first image the renderer generates).

https://github.com/etlegacy/etlegacy/blob/b6e0e88c501f1a7cbe508b8a531a99c07f4136f5/src/renderer/tr_image.c#L766-L779

All variables up until this point seem identical to when compiling with SDL2.0.20, it simply gets `SIGSEGV` signal when calling this function.

**Comment by Aciz** [comment]
Looks like this might be an issue with GLEW 2.2.0 rather than SDL2.0.22. I downgraded to 2.0.20 earlier today, but the issue persisted - however GLEW was also at newer version from what we ship (2.2.0 local, 2.1.0 shipping). Need to investigate further if it's actually GLEW issue, but seeing as [ETe](https://github.com/etfdevs/ETe) for example compiles and runs fine with 2.0.22, I am leaning towards GLEW issue rather than SDL (ETe doesn't use GLEW).

**Comment by biopsin** [comment]
I can test it hopefully this sunday as Im on Glew-2.2.0 aswell, downgrade it and update SDL2-2.0.22 and confirm.

**Comment by Aciz** [comment]
Eh, testing now again and seems it is SDL issue after all. I have both SDL2.0.22 as well as GLEW 2.2.0 installed, and compiling latest master with:
* `-DBUNDLED_GLEW=OFF` = works (SDL 2.0.20 + GLEW 2.2.0)
* `-DBUNDLED_SDL=OFF` = segfault (SLD 2.0.22 + GLEW 2.1.0)
* `-DUNDLED_GLEW=OFF -DBUNDLED_SDL=OFF` = segfault (SDL 2.0.22 + GLEW 2.2.0)

**Comment by biopsin** [comment]
Currently testing SDL2 2.24.0 with --disable-render as configure option. Even tho I cant see in the configurelog any confirmastion that render is disable, ETlegacy runs fine.
Will latert test default SDL2 build with --enable-render

**Comment by Aciz** [comment]
Fixed as part of 5935e4e52d8c7c69201ca3ce448170d7dece2fe5, the splash screen itself is not functional but it no longer segfaults.
